### DISCUSSION TITLE
Adobe Air application  installation on sonoma issue

### DISCUSSION THREAD

**Discussion by cristallium** [airsdk/Adobe-Runtime-Support#2939] [Q&A]
Hi,

I have a old Adobe air application that doesn't want to be run on sonoma (others versions of mac OS are fine).

If I try running app from terminal, here is the log : 

The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x600003789440 {Error Domain=NSPOSIXErrorDomain Code=153 "Unknown error: 153" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}
mycomputer@gdg /Applications %

I have tried with Adobe Air runtime 32 and AIR 50.2.3.8, same result.

Any idea of what is the issue ?

regards

**Comment by ajwfrost** [comment]
How are you trying to run it? Is it a .air file that you've installed (and if so .. what version of the AIR runtime is installed on your machine? - 50.2.3.8 is an AIR SDK version but there was no runtime release with that number). Or are you trying to use ADL to launch it?

The most likely issue would be related to code-signing, although I just tried using an old (public) .air file along with AIR runtime 33.1.1.744 on Sonoma 14.1.1 and that one launched fine. But then it would be an x86_64 binary and there are different rules for that (but then, the same should apply to your attempt with AIR runtime 32).

And do you know whether the application requires the webkit capability i.e. does it use HTML-based content or the HTMLLoader class? (Is it possible to provide details or a link for this AIR-based app so that we can try to reproduce it?)

thanks

**Reply by cristallium** [reply]
Hi, thanks for your reply. 
The file application is an .air file compiled with FlashBuilder4.7 using Flex4.6.0 SDK (yes it's very old). 
This is running fine on PC with Air runtime 32 or 50 installed on the machine. But on sonama, it fail. 

Search for "HTMLLoader" in the code return 0 result. But I found some Text components like this : `<mx:Text htmlText="some text" color="#002B6B"/>`. Is it using webkit ?

How to know all componants using webkit ? This could be the source of the issue ?

**Reply by ajwfrost** [reply]
So it's more possible that you don't need webkit - actually, if you're running it on Windows with AIR runtime 50, then it definitely doesn't need webkit.

Which means that the problem might be that you've got an arm64 binary being generated for the application on Sonoma, and potentially Sonoma is blocking this because it's not code-signed.

A couple of things to check:

1) if you open a terminal window, can you copy the response to the below commands for us?
   - `codesign -v -vvv /Applications/yourappname.app`
   - `codesign -d -vvv /Applications/yourappname.app`
("yourappname" will be whatever you see in Finder when you look for this in the Applications folder)

2) After trying to open it, can you go into System Settings and open the Privacy & Security, section, then scroll right down.. is there anything there about an app trying to open that's not from an identified developer?

3) If you open the "Console" app (under /Applications/Utilities), can you click on "Crash Reports" and see if you can find one relating to this app? If so, please copy/paste the report text from the lower part of that window.


thanks

**Reply by cristallium** [reply]
Hi Andrew,
Thanks for your reply. I sent you the air file by PM. I'll try this as soon I have access to a Mac with sonoma

**Reply by ajwfrost** [reply]
Hi -> I've not received the .air file, not quite sure how you'd sent that but if email servers are involved, it may have been blocked.. in case it helps, here's an upload link:
https://transfer.harman.com/requests/NfdqFTCcS8P3ma1BmX3hkl
thanks

**Reply by cristallium** [reply]
Effectively, I got a return mail. I send it on your public email github profile. 
Is the link provided private ?

**Reply by ajwfrost** [reply]
Ah, I'll have to check whether that still works! Old company domain name, it was meant to forward on to the harman.com one, but they may have changed that now. Or it may just have been too large/etc...
Yes the link above would mean an upload that only I see, no one else clicking on it will see what's been transfered.
thanks

**Reply by ajwfrost** [reply]
Okay so interestingly, if I run this from Finder, I get a message "This application cannot be opened".. although if I run it from the command line using `/Applications/appname.app/Contents/MacOS/appname` then it does open!

A better solution though: if you right-click the app in Finder, and choose "Get Info", you will hopefully see a checkbox option near the top of the info box that allows you to set it to open using Rosetta. This seems to then help, because the security requirements are lowered for this option!

Longer term .. there's not a huge amount we can do here I think, given how the application executables are generated from a template and there's no chance to properly sign the package (as it needs the certificates etc to be present on the signing machine..). But we'll have a think about options... definitely better if companies can distribute signed bundles rather than .air files now..!

thanks

**Reply by cristallium** [reply]
Thanks for your quick investigation and big thanks for (provisory) solution.
I know how to make an .exe from .airi for PC modifying the application descriptor  with `<supportedProfiles>desktop extendedDesktop </supportedProfiles>` 
and after, I sign it with adt
But for Mac, is it the same process ? Do I need an Apple dev certificate ou just my code signing certificate ? Is there some tutos somewhere ? What I understand is that I need to use a Mac OS with sonoma, no way to cross compile... :-(
I suppose, I also need to make an Mac native updater to update the application ?

**Reply by ajwfrost** [reply]
Ah, if you're developing the app itself, then yes you would be able to create the bundle on macOS as well... but yes there is no option for cross-compiling this so you would need to have a macOS computer, and an Apple developer account so that you can create the "Developer ID Application" certificate that needs to be used to sign the app (and then if you're creating an installer for it, a "Developer ID Installation" certificate... and notarization then so that you can distribute it to people and they can see that Apple have already inspected the software..)

Some guidelines:
https://airsdk.dev/docs/tutorials/platform/macos/creating-macos-pkg-files

thanks
